explorer/interpreter/heap.h

```diff
@@ -19,7 +19,7 @@
 namespace Carbon {
 
 // A Heap represents the abstract machine's dynamically allocated memory.
-class Heap : public HeapAllocationInterface {
+class Heap : public Printable<Heap, HeapAllocationInterface> {
```

Comment:
```suggestion
class Heap : public HeapAllocationInterface>, public Printable<Heap> {
```
Can we directly inherit instead of chaining through `Printable` here? That'd be easier to reason about, despite introducing multiple inheritance. (I think this doesn't violate the ideal of doing things like we would in Carbon, because in Carbon we'd model `Printable` as a mixin.)

Reply:
Yes, I just hadn't undone this case. For me, there's a question of how valuable consistency would be.

(note there's a typo in your suggestion, extra `>`)